skill-tree /src/main/java/com/RDS/skilltree/utils/JWTUtils.java

```diff
@@ -0,0 +1,79 @@
+package com.RDS.skilltree.utils;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+
+@Component
+public class JWTUtils {
+
+    @Value("${jwt.rds.public.key}")
+    private String publicRDSKeyString;
+
+    /**
+     * Converts the given private key string to an RSAPrivateKey object.
+     *
+     * @param privateKeyString The private key string to be converted.
+     * @return The corresponding RSAPrivateKey object.
+     * @throws Exception If an error occurs during the conversion process.
+     */
+    private RSAPrivateKey convertToRSAPrivateKey(String privateKeyString) throws Exception {
```

Comment:
Tried this something like this, but got error 
```
try (KeyFactory keyFactory = KeyFactory.getInstance("RSA"){
        publicKey = keyFactory.generatePublicKey(keySpec);
}
```

The KeyFactory class is not an Autocloseable class, so we will not require the try-with-resources, have handled the NoSuchAlgo exception seperately instead of throwing the `Invalid public key exception`

Reply:
This we cannot use with try-with-resources as KeyFactory doesn't implement the AutoCloseable